pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -405,6 +405,8 @@ func Run(ps *pubsub.PubSub) {
 			subAssignableAdapters.ProcessChange(change)
 
 		case change := <-subAppNetworkConfig.MsgChan():
+			// If we have an NetworkInstanceConfig process it first
+			checkAndProcessNetworkInstanceConfig(&zedrouterCtx)
```

Comment:
Hmm.. Very likely - NIC will come right after this notification about ANC - right?
Should ANC handler just check and return and NIC detect that it needs to kick stalled AIC processing ? Wouldn't that be a better model instead of insisting on the order?

Reply:
We know that zedagent publishes the NeworkInstanceConfig before it publishes the AppInstanceConfig, and the latter is then processed by zedmanager before zedrouter woll see an AppNetworkConfig. But since the select construct has zero fairness in golang we have no idea when these handlers will be called. But the goroutine in pubsub is getting the data over AF_UNIX.

The reason I don't want to have the code silently wait for the ANC and NIC forever is that it means zero error reporting (a silent wait forever in Init state) when the controller has forgotten to send the NIC. We'll see if this closes the race sufficiently for it not to appear in the test runs and in normal usage.

Reply:
So still a bit of an experiment. But takes forever to verify since each test run takes 4 hours or more.